### PR TITLE
1.5.0

### PR DESCRIPTION
**Pull request by tab-cmd**
# Version 1.5.0

Although the branch is named 1.4.3, we changed quite a bit and warranted a larger change to 1.5.0! This PR represents a major undertaking for the group and we're excited to release it to the world. We already have plans to release 1.5.1 with even more refactors and tooling to make development in BciPy easier! We'll also start adding to our official website at bcipy.github.io.

## Contributions

There were many contributions, including small bug fixes. We'll describe the major ones below.

### Added
- Language model histogram #91 
- BciPy official glossary (Sequence -> Inquiry & Epoch -> Series) #121 
- System information to `system_utils` (cpu, platform, etc) #98 
- BciPy Experiments and Fields: See PRs #113 #111 and #114 for more information on the additions!
- `.bcipy` system directory to support experiment and fields #100 
- support for python 3.7
- `rsvp/query_mechanisms`: to model the way we build inquiries #108 
- `Makefile`: contains useful install and development commands
- `convert`: a module for data conversions that will be useful for data sharing. Implemented a conversion function to the EDF format. #104 
- `exceptions`: a module for BciPy core exceptions

### Updated
- `acquisition`: refactored the acquisition module to separate the concept of a device (ex. DSI-24 headset) and a connection method to that device (TCP or LSL). #122 
- `setup.py`: with new repo location and CAMBI official support email 
- `offline_analysis`: to pull parameters from session file #90 
- `requirements.txt`: to the latest available #99 #107 
- `Parameters` (added help text, removed redundant parameters). Refactored to make them immutable. #101 
- `gui_main `: to use PyQt5. We will refactor all GUI code to use this in the future. After this PR, the signal viewer (WxPython) and a couple of loading functions will remain (Tk). #102 
- `BCInterface` : updated to use new gui_main methods. Added user if validations. #102  #120 
- `params_form`: moved into a parameters modules within GUI and updated to use PyQt5. #109 
- `dev_requirements`: used to be called test_requirements. It contains more than that, so we updated the name! #99 
- `README`: with relevant updates and contributors 

### Removed
- `RSVPKeyboard.py`

## Testing

- [x] `make test-all`: all tests pass
- [x] RSVP calibration : real data : ensure adequate AUC: .93 acheived. 
- [x] RSVP Copy phrase: real data: ensure spelling can occur. Success.
- [x] Run all registered tasks (except icon tasks)

*Note: because no language model changes occurred it was not tested with language model enabled.*

## Documentatation

Updated relevant READMEs and docstrings.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -12,10 +10,10 @@
 DESCRIPTION = 'Python Software for Brain-Computer Interface.'
 URL = 'https://github.com/BciPy/BciPy'
 EMAIL = 'memmott@ohsu.com'
```
Comment (tab-cmd) [inline setup.py]: update to cambi support email

File: setup.py [hunk 2]
```diff
@@ -12,10 +10,10 @@
 DESCRIPTION = 'Python Software for Brain-Computer Interface.'
 URL = 'https://github.com/BciPy/BciPy'
```
Comment (tab-cmd) [inline setup.py]: update url!

**Review by lawhead — DISMISSED** [review]

File: bcipy/acquisition/README.md [hunk 3]
```diff
@@ -30,23 +30,32 @@ The daq/tests/client_test.py file also demonstrates how the various components i
...
-The registry has a list of supported devices. Devices can be queried by name. The device constructor is returned.
+The registry has a list of supported connectors, which can be queried by device and connection method.
 
     import daq.protocols.registry as registry
-    Device = registry.find_device('DSI')
+    registry.find_connector(device_spec=supported_device('DSI'), 
+                            connection_method=ConnectionMethod.TCP)
+
+
+#### Connector
```
Comment (celikbasak) [inline bcipy/acquisition/README.md:46]: Sorry for the novice question, but what type of connector we are using in the system? Should we give an example here, or would defining params in configure_connector be enough?

File: bcipy/acquisition/demo/demo_client_and_server.py [hunk 4]
```diff
@@ -19,49 +19,60 @@ def main():
...
+    from bcipy.acquisition.devices import supported_device
+    from bcipy.acquisition.connection_method import ConnectionMethod
+    from bcipy.helpers.system_utils import log_to_stdout
+
+    if debug:
+        log_to_stdout()
 
     host = '127.0.0.1'
     port = 9000
     # The Protocol is for mocking data.
```
Comment (celikbasak) [inline bcipy/acquisition/demo/demo_client_and_server.py]: Comment is a little bit unclear - is it to mock the data or is it using the mocked data?

File: bcipy/acquisition/devices.json [hunk 5]
```diff
@@ -0,0 +1,54 @@
+[
+    {
```
Comment (celikbasak) [inline bcipy/acquisition/devices.json:2]: This is a very useful json file for devices and parameters. Thank you!

**Review by AlisterD — DISMISSED** [review]

**Review by celikbasak — DISMISSED** [review]

File: bcipy/acquisition/demo/demo_client_and_server.py (hunk 4 above)
Comment (lawhead) [reply bcipy/acquisition/demo/demo_client_and_server.py]: I refactored this code but forgot to remove this comment. It's not relevant anymore so I'll delete it. Sorry for the confusion.

File: bcipy/acquisition/README.md (hunk 3 above)
Comment (lawhead) [reply bcipy/acquisition/README.md:46]: @celikbasak, it's helpful to know which parts of the documentation are unclear. I updated the README with an additional paragraph for currently supported Connectors. Hopefully this helps.

**Review by celikbasak — APPROVED** [review]

**Review by lawhead — APPROVED** [review]

**Review by AlisterD — APPROVED** [review]

**Pull request merged**